tests/lint/unittest_lint.py

```diff
@@ -525,6 +525,183 @@ def test_load_plugin_command_line() -> None:
+    """
+    Case 3 refers to issue #7264.
+    This is where we supply a plugin we want to load on both the CLI only,
+    but that plugin is only loadable after the ``init-hook`` in
+    the config file has run. This is not supported, and was previously a silent
+    failure. This test ensures a ``bad-plugin-value`` message is emitted.
+    """
+    dummy_plugin_path = abspath(join(REGRTEST_DATA_DIR, "dummy_plugin"))
+    with fake_home() as home_path:
+        # construct a basic rc file that just modifies the path
+        pylintrc_file = join(home_path, "pylintrc")
+        with open(pylintrc_file, "w", encoding="utf8") as out:
+            out.writelines(
+                [
+                    "[MASTER]\n",
+                    f"init-hook=\"import sys; sys.path.append(r'{home_path}')\"\n",
+                ]
+            )
+
+        copytree(dummy_plugin_path, join(home_path, "copy_dummy"))
+
+        # To confirm we won't load this module _without_ the init hook running.
+        assert home_path not in sys.path
+
+        run = Run(
+            [
+                "--rcfile",
+                pylintrc_file,
+                "--load-plugins",
+                "copy_dummy",
+                join(REGRTEST_DATA_DIR, "empty.py"),
+            ],
+            reporter=testutils.GenericTestReporter(),
+            exit=False,
+        )
+        assert run._rcfile == pylintrc_file
+        assert home_path in sys.path
+        # The module should not be loaded
+        assert not any(ch.name == "copy_dummy" for ch in run.linter.get_checkers())
+
+        # There should be a bad-plugin-message for this module
+        assert len(run.linter.reporter.messages) == 1
+        assert run.linter.reporter.messages[0] == Message(
+            msg_id="E0013",
+            symbol="bad-plugin-value",
+            msg="Plugin 'copy_dummy' is impossible to load, is it installed ? ('No module named 'copy_dummy'')",
+            confidence=interfaces.Confidence(
+                name="UNDEFINED",
+                description="Warning without any associated confidence level.",
+            ),
+            location=MessageLocationTuple(
+                abspath="Command line or configuration file",
+                path="Command line or configuration file",
+                module="Command line or configuration file",
+                obj="",
+                line=1,
+                column=0,
+                end_line=None,
+                end_column=None,
+            ),
+        )
+
+        sys.path.remove(home_path)
+
+
+def test_load_plugin_command_line_before_init_hook() -> None:
+    regrtest_data_dir_abs = abspath(REGRTEST_DATA_DIR)
+
+    run = Run(
+        [
+            "--load-plugins",
+            "dummy_plugin",
+            "--init-hook",
+            f'import sys; sys.path.append("{regrtest_data_dir_abs}")',
+            join(REGRTEST_DATA_DIR, "empty.py"),
+        ],
+        exit=False,
+    )
+    assert (
```

Comment:
Why is this 2? 😕

Reply:
I do not know. I copied this wholesale from an existing test, `test_load_plugin_command_line`, which is [here](https://github.com/PyCQA/pylint/blob/8b46e222b4d8f3abc01c9bde58e4ecc5af617fef/tests/lint/unittest_lint.py#L512-L525)
If I had to guess, I would imagine it is because the Dummy Plugin file supplies two checkers for other test purposes: https://github.com/daogilvie/pylint/blob/be4df46ab61d233e7e18ce0b3f4b8491b8138893/tests/regrtest_data/dummy_plugin/dummy_plugin.py#L5-L28

Reply:
Makes sense. Sorry should have checked that myself!